jfrog-cli/utils/tests/utils.go

```diff
@@ -308,3 +316,84 @@ func GetBaseDir(setToParentDir bool) (baseDir string) {
 	baseDir = filepath.Join(pwd, "testdata")
 	return
 }
+
+func GetNonVirtualRepositories() map[string]string {
+	return map[string]string{
+		Repo1:             SpecsTestRepositoryConfig,
+		Repo2:             MoveRepositoryConfig,
+		LfsRepo:           GitLfsTestRepositoryConfig,
+		JcenterRemoteRepo: JcenterRemoteRepositoryConfig,
+		NpmLocalRepo:      NpmLocalRepositoryConfig,
+		NpmRemoteRepo:     NpmRemoteRepositoryConfig,
+	}
+}
+
+func GetVirtualRepositories() map[string]string {
+	return map[string]string{
+		VirtualRepo: VirtualRepositoryConfig,
+	}
+}
+
+func getRepositoriesNameMap() map[string]string {
+	return map[string]string{
+		"${REPO1}":             Repo1,
+		"${REPO2}":             Repo2,
+		"${VirtualRepo}":       VirtualRepo,
+		"${LfsRepo}":           LfsRepo,
+		"${JcenterRemoteRepo}": JcenterRemoteRepo,
+		"${NpmLocalRepo}":      NpmLocalRepo,
+		"${NpmRemoteRepo}":     NpmRemoteRepo,
+		"${GoRepo}":            GoLocalRepo,
+	}
+}
+
+func ReplaceTemplateVariables(path, destPath string, replaceCredentials bool) (string, error) {
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	repos := getRepositoriesNameMap()
+	for repoName, repoValue := range repos {
+		content = bytes.Replace(content, []byte(repoName), []byte(repoValue), -1)
+	}
+	if destPath == "" {
+		destPath, err = os.Getwd()
+		if err != nil {
+			return "", errorutils.CheckError(err)
+		}
+		destPath = filepath.Join(destPath, Temp)
+	}
+	err = os.MkdirAll(destPath, 0700)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	template := filepath.Join(destPath, filepath.Base(path))
+	log.Debug("Writing content information to:", template)
+	if replaceCredentials {
+		for authenticationName, authenticationValue := range getAuthenticationMap() {
+			content = bytes.Replace(content, []byte(authenticationName), []byte(authenticationValue), -1)
+		}
+	}
+
+	err = ioutil.WriteFile(template, []byte(content), 0700)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	// Now that the content is good we need to write the output to a file.
+	return template, nil
+}
+
+func PreparePath(fileName string) (string, error) {
```

Comment:
I suggest renaming the function to CreateSpec.

Reply:
Done